docs/proposals/kubelet-rootfs-distribution.md

```diff
@@ -0,0 +1,163 @@
+
+However, right now it's not always easy to run it *correctly*. The Kubelet has
+a number of dependencies that must exist in its filesystem, including various
+mount and network utilities. Missing any of these can lead to unexpected
+differences between Kubernetes hosts. For example, the Google Container VM
+image (GCI) is missing various mount commands even though the Kernel supports
+those filesystem types. Similarly, CoreOS Linux intentionally doesn't ship with
+many mount utilities or socat in the base image.
+
+In order to solve this problem, it's proposed that running the Kubelet in a
+prepackaged chroot should be a supported, recommended, way of running a fully
+functioning Kubelet.
+
+## The Kubelet Chroot
+
+The easiest way to express all filesystem dependencies of the Kubelet comprehensively is to ship a filesystem image and run the Kubelet within it. The [hyperkube image](../../cluster/images/hyperkube/) already provides such a filesystem.
+
+Even though the hyperkube image is distributed as a container, this method of
+running the Kubelet intentionally is using a chroot and is neither a container nor pod.
+
+The kubelet chroot will essentially operate as follows:
+
+```
+container-download-and-extract gcr.io/google_containers/hyperkube:v1.4.0 /path/to/kubelet
+mount --make-shared /var/lib/kubelet
+mount --rbind /var/lib/kubelet /path/to/kubelet/var/lib/kubelet
+mount --rbind /sys /path/to/kubelet/sys
+...
+chroot /path/to/kubelet /usr/bin/hyperkube kubelet
+```
+
+Note: Kubelet might need access to more directories on the host and we intend to identity mount all those directories into the chroot. A partial list can be found in the CoreOS kubelet-wrapper script.
+This logic will also naturally be abstracted so it's no more difficult for the user to run the Kubelet.
+
+## Current Use
+
+This method of running the Kubelet is already in use by users of CoreOS Linux. The details of this implementation are found in the [kubelet wrapper documentation](https://coreos.com/kubernetes/docs/latest/kubelet-wrapper.html).
+
+## Implementation
+
+### Target Distros
+
+The two distros which benefit the most from this change are GCI and CoreOS. Initially, these changes will only be implemented for those distros.
+
+This work will also only initially target the GCE provider and `kube-up` method of deployment.
+
+#### Hyperkube Image Packaging
+
+The Hyperkube image is distributed as part of an official release to the `gcr.io/google_containers` registry, but is not included along with the `kube-up` artifacts used for deployment.
+
+This will need to be remediated in order to complete this proposal.
+
+#### Everything else
+
+In the initial implementation, rkt or docker can be used to extract the rootfs of the hyperkube image. rkt fly or a systemd unit (using `RootDirectory`) can be used to perform the needed setup, chroot, and execution of the kubelet within that rootfs.
+
```

Comment:
Add a testing and rollout plan. The way I see it, we will have to identify & clone a subset of high value e2e tests to run kubelet in a chroot. Once we gain confidence, we need to enable this across all e2es. 
Given that there is only one feature change required for this feature, we should target v1.6.